### DISCUSSION TITLE
Simple tutorial

### DISCUSSION THREAD

**Discussion by CesarSilvaCP** [TomorrowIdeas/plaid-sdk-php#40] [Q&A]
Hi! Glad to find a PHP version!

I'm trying to do the first task, this is my code:

======
$plaid = new Plaid(
'.....',
'...',
'sandbox'
);

$token = $plaid->tokens->create("Plaid App", "es", ["US","CA"], $user_id,["auth"]);

=====

But I'm having the following error:

**Fatal error: Uncaught TypeError: Argument 4 passed to TomorrowIdeas\Plaid\Resources\Tokens::create() must be an instance of TomorrowIdeas\Plaid\Entities\User, null given, called in C:\xampp\htdocs\agint\index.php on line 22 and defined in C:\xampp\htdocs\agint\vendor\tomorrow-ideas\plaid-sdk-php\src\Resources\Tokens.php:27 Stack trace: #0 C:\xampp\htdocs\agint\index.php(22): TomorrowIdeas\Plaid\Resources\Tokens->create('Plaid App', 'es', Array, NULL, Array) #1 C:\xampp\htdocs\agint\index.php(26): CalTok() #2 {main} thrown in C:\xampp\htdocs\agint\vendor\tomorrow-ideas\plaid-sdk-php\src\Resources\Tokens.php on line 27**

======

I'm certainly doing something wrong, I'm searching google if I find any examples using this repository, but I haven't found any examples.

==

Are there any examples available?

**Comment by syphen** [comment]
So, the api is looking for a user object and you are trying to pass the $user_id instead. You might need to do this first:
`$user = new \TomorrowIdeas\Plaid\Entities\User($user_id);`

And then your call:
`$token = $plaid->tokens->create("Plaid App", "es", ["US","CA"], $user,["auth"]);`

**Reply by CesarSilvaCP** [reply]
Hi thanks!
The error is gone.

Thank you for your help.
-----

I recently got to know this API, I'm getting a bit confused with it :)

----

**Reply by CesarSilvaCP** [reply]
Would the next task be the following instruction?
$user = new \TomorrowIdeas\Plaid\Entities\User($user_id);

$token = $plaid->tokens->create("Plaid App", "es", ["US","CA"], $user,["auth"]);
$plaid->tokens->get($token);

---
If you can help me, thanks in advance.

**Reply by syphen** [reply]
What are you trying to do exactly?

**Reply by CesarSilvaCP** [reply]
I'm trying to recreate the demo. The task of opening the modal and confirming the use of the product[auth]
The Code at the moment:
`<?php 
use TomorrowIdeas\Plaid\Plaid;

require __DIR__ . "/vendor/autoload.php";

function CalTok(){
	$plaid = new Plaid(
		'---------------',
		'---------------',
		'sandbox'
	);
	
	// $user_id = '---------------';

	$user_id = '4514';
	$user = new \TomorrowIdeas\Plaid\Entities\User($user_id);

	$token = $plaid->tokens->create("Plaid App", "es", ["US","CA"], $user,["auth"]);
	// $auth = $plaid->auth->get($token);
	// return $plaid;
}

$plaid = new Plaid(
	'---------------',
	'---------------',
	'sandbox'
);

$user_id = '4514';
$user = new \TomorrowIdeas\Plaid\Entities\User($user_id);

$token = $plaid->tokens->create("Plaid App", "es", ["US","CA"], $user,["auth"]);

`

**Reply by syphen** [reply]
Ahh. So that actually requires a combination of server-side and client-side code. You'll need to use the javascript library for Plaid Link ([https://plaid.com/docs/link/web/](https://plaid.com/docs/link/web/)) to get a user token to then use with this PHP library.

**Reply by syphen** [reply]
Also...don't put your keys up on git.

**Reply by CesarSilvaCP** [reply]
Oh yes, now I understand.
Thank you very much for the clarification.
